### PR TITLE
(#140) multiple category

### PR DESCRIPTION
**Pull request by pankaj-ag**
PR for  #140

Now you can select multiple categories.

### CODE REVIEW

File: resources/assets/js/app.js [hunk 1]
```diff
@@ -396,18 +396,48 @@ if (document.getElementById('books_listing')) {
...
+                            id:checkbox.value
+                        }); 
+                    }
+                });
 
-            updateCategory: function(index, bookID) {
+                this.$set(this.books[this.currentBookIndex], 'categories',  selectedCategory);
                 let route = `${this.updateRoute}/${bookID}`;
-                axios.put(route, {categories:this.books[index]['categories']});
+                axios.put(route, {categories: JSON.parse(JSON.stringify(selectedCategory))});
```
Comment (pankaj-ag) [inline resources/assets/js/app.js]: Converting JS array to JSON.

File: resources/assets/js/app.js [hunk 2]
```diff
@@ -396,18 +396,47 @@ if (document.getElementById('books_listing')) {
...
-            updateCategoryMode : function(index, mode) {
-                this.$set(this.books[index], 'showCategories', (mode === 'edit'));
+            updateCategoryMode : function(index) {
+                let categories = this.books[index]['categories'];
+                if(!categories) {
+                    return false;
+                }
+                this.currentBookIndex = index;
+
+                this.categoryInputs.map((checkbox) => checkbox.checked = false );
```
Comment (rathorevaibhav) [inline resources/assets/js/app.js]: Not very sure if we've completely utilized VueJS here. As far as I could read, we're only using vue for detecting on-click events. Maybe this checkboxes handling can also be implemented using Vue.

Again not sure, but we should definitely look on this. @pankaj-ag thoughts?

File: resources/assets/js/app.js [hunk 3]
```diff
@@ -396,18 +396,47 @@ if (document.getElementById('books_listing')) {
...
+                categories.forEach((category) => this.categoryInputs[category.id].checked =  true );
```
Comment (rathorevaibhav) [inline resources/assets/js/app.js]: 👍 for ES7

File: resources/assets/js/app.js (hunk 2 above)
Comment (pankaj-ag) [reply resources/assets/js/app.js]: Great point @rathorevaibhav. 
I am using this variable for some other data also (category name ) but I'll have a look on this and try to do this with modal binding in next PR.

**Comment by rathorevaibhav** [comment]
Checked this PR :

- [ ] For duplication.
- [ ] For performance. 
- [ ] Usage of recursive methods, loops, and algorithms.
- [ ] For readability and understanding. 
- [ ] For functions with multiple responsibilities.
- [ ] For scope of refactoring. 
- [ ] For coupling
- [ ] For alternative approaches to the implementation. 
- [ ] For how the change impacts other parts of the system.
- [ ] For validations. client side and server side validations are present.

For more insights on code reviews checkout our [guidelines](https://docs.google.com/document/d/1Sa-XYJEmGwrOM5gk5zoan_Z6j90ylibmCmt331361nw/edit#).

**Pull request merged**